torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1266,17 +1266,32 @@ void ProcessGroupNCCL::heartbeatMonitor() {
               coordCheckIntervalMilSec_) {
         lastTimePollStore = currentTime;
         if (globalStore_->check({std::string(TIMEOUT_DUMP)})) {
+          int timeOutRank = -1;
+          try {
+            auto vec = globalStore_->get(std::string(TIMEOUT_DUMP));
+            TORCH_CHECK_WITH(
+                DistBackendError,
+                vec.size() == sizeof(int),
+                "Invalid size for the timeout rank ID");
+            std::memcpy(&timeOutRank, vec.data(), vec.size());
+          } catch (const std::exception& e) {
+            LOG(ERROR)
+                << "Failed to get timeout rank ID from the global store.";
+          }
           errorMsg = c10::str(
               logPrefix(),
-              "Received a global timeout from another rank and will ",
-              "start to dump the debug info. ",
+              "Received a global timeout from another rank ",
```

Comment:
nit:

"another rank {rank}" doesn't sound as good as just "rank {rank}"   (can delete 'another')

also, i think you could add "signal"

"Received a global timeout signal from rank {rank}" sounds better to me